src/librustc_typeck/check/mod.rs

```diff
@@ -4344,7 +4344,11 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                                 let max_len = receiver.rfind(".").unwrap();
                                 format!("{}{}", &receiver[..max_len], method_call)
                             } else {
-                                format!("{}{}", receiver, method_call)
+                                match &expr.kind {
+                                    ExprKind::Binary(_,_,_) => format!("({}){}", receiver, method_call),
+                                    ExprKind::Unary(_,_) => format!("({}){}", receiver, method_call),
+                                    _ => format!("{}{}", receiver, method_call),
+                                }
```

Comment:
Use `expr.precedence().order() < ExprPrecedence::MethodCall.order()` instead of matching on the `kind`, that way this will never be incorrect for any expression type, even if we add new ones.

Reply:
@estebank I made the change. But it seems to break/change test cases for other scenarios too.

Reply:
That is fine, it means those suggestions were incorrect in the first place.

Reply:
Cool